src/instructlab/model/dk_bench_utils.py

```diff
@@ -0,0 +1,408 @@
+        elif model_path.is_file():
+            valid_model = is_model_gguf(model_path)
+    return valid_model
+
+
+def get_endpoint_model_name(openai_client: OpenAI, model_name: str | None) -> str:
+    try:
+        models = openai_client.models.list()
+    except OpenAIError as exc:
+        raise ClientException(
+            f"Connection Error {exc}. Unable to list models at"
+        ) from exc
+
+    endpoint_model_names = [model.id for model in models.data]
+
+    if model_name is not None:
+        if model_name not in endpoint_model_names:
+            raise ValueError(
+                f"Model named {model_name} is not listed at endpoint. Models listed at endpoint are: {endpoint_model_names}"
+            )
+    # No model name input by user
+    else:
+        if len(endpoint_model_names) == 1:
+            model_name = endpoint_model_names[0]
+        else:
+            raise ValueError(
+                f"More than one model listed at endpoint {openai_client.base_url}. To evaluate a model at this endpoint pass in one of following models listed: {endpoint_model_names}"
+            )
+
+    logger.debug("Model %s found at endpoint %s", model_name, openai_client.base_url)
+    return model_name
+
+
+def create_results_file_name(
+    file_format: str, output_dir: str, timestamp: str, model_name: str
+) -> str:
+    if IOFileType.CSV.value == file_format:
+        file_type = "csv"
+    elif IOFileType.XLSX.value == file_format:
+        file_type = "xlsx"
+    elif IOFileType.JSONL.value == file_format:
+        file_type = "jsonl"
+    else:
+        raise ValueError("File format is not one of: csv, xlsx, jsonl")
+
+    output_path = pathlib.Path(output_dir)
+    if not output_path.exists() or not output_path.is_dir():
+        output_dir = DEFAULTS.EVAL_DATA_DIR
+        logger.debug(
+            "Output dir provided is not a directory or does not exist, writing results to default evaluation output directory %s",
+            output_dir,
+        )
+
+    return f"{output_dir}/responses-{model_name}-{timestamp}.{file_type}"
+
+
+def make_run_dir(output_dir: str) -> str:
+    now = datetime.now()
+    timestamp = now.isoformat()
+    # remove any training slashes
+    output_dir = os.path.normpath(output_dir)
+    run_dir = f"{output_dir}/job-{timestamp}"
+
+    path_does_not_exist = not os.path.exists(output_dir)
+    path_is_not_dir = not os.path.isdir(output_dir)
+    if path_does_not_exist or path_is_not_dir:
+        run_dir = f"{DEFAULTS.EVAL_DATA_DIR}/job-{timestamp}"
+        logger.debug(
+            "User provided output directory path %s does not exist or is not a directory. Creating directory for results in %s",
+            output_dir,
+            DEFAULTS.EVAL_DATA_DIR,
+        )
+
+    os.makedirs(run_dir, exist_ok=True)
+    logger.debug("Created directory for results at %s", run_dir)
+    return run_dir
+
+
+def print_results(result: EvaluationResult, results_files: List[str], model_name: str):
+    print("\n")
```

Comment:
Is there a reason why we're using `print` statements instead of the `logger`?

Reply:
Right now each of the benchmarks produces an output of scores with print statements. Take a look at this example: https://github.com/instructlab/instructlab/blob/main/src/instructlab/model/evaluate.py#L144

I'm using prints here to be consistent with the output of the other benchmarks. We could revisit the use of print vs logger in the future for the outputs of these commands.

Reply:
I think the right way to look at this (at least as I understand it) is logger should be viewed as optional information at different levels of granularity.  We should view the logger as something that could be turned off entirely or sent to a separate log file. The print statements are not optional (although we should eventually have the option to have different formats (Ex: json)).  The point of print statements is to give a pretty formatted response from the command.  As an alternative to print, we could be using click.echo which serves the same purpose of controlling the output.

Reply:
@cdoern Any thoughts here?

Reply:
I think using print is totally fine for things meant to be formatted a certain way (like Eval output) since the logger kind of ruins that. 

In general we should use the logger everywhere we can, but this seems to be an exception perhaps except the `"Responses and scores written to:"` line

Reply:
The idea I had behind giving the user the path is that if they wanted to dig into the score for a specific question, they could open up one of the response files and look at it immediately.

It would look funny in the code but I don't think it would be a bad idea to have the same message set at level `DEBUG` as well.

Reply:
> It would look funny in the code but I don't think it would be a bad idea to have the same message set at level `DEBUG` as well.

We probably should be doing this in a lot of places.  Again since the logs need to be able to live on their own.

Reply:
Thinking about the logger vs print/echo separation a bit more, I think part of the issue here is we now have a new rule with the click separation.  The rule should probably be that if you don't have access to click, you can't print.  Or just enforce as don't use print and use click.echo instead.  But that logic hasn't been implemented fully with the click separation.  With eval for example, what should probably be happening is when the eval is finished, the results should be passed back to the cli eval click impl and it should be doing the click.echoing of the results.  Instead it's implemented as a fire and forget.  I might be misunderstanding the new logic, but I think this is what's needed to have a client agnostic impl going forward.

Reply:
Agreed, `evaluate_model()` should be returning some kind of structure that we can use to create the output in `cli/evaluate.py`. Seems like a good follow up issue to be worked on after this is merged.